pkg/internal/server/grpc/snaphotmetadata_handlers.go

```diff
@@ -36,16 +43,52 @@ func (s *Server) GetMetadataAllocated(req *api.GetMetadataAllocatedRequest, stre
 		return err
 	}
 
-	// TODO: Call CSI driver endpoint for changed block metadata
+	csiClient := csi.NewSnapshotMetadataClient(s.csiConnection())
+	csiReq := newCSIGetMetadataAllocatedRequest(ctx, snapshotHandle, req)
+
+	// Invoke CSI Driver's GetMetadataDelta gRPC and stream the response back to client
+	csiStream, err := csiClient.GetMetadataAllocated(ctx, csiReq)
+	if err != nil {
+		return err
+	}
+
+	done := make(chan bool)
+	errCh := make(chan error, 1)
+	go func() {
+		for {
+			csiResp, err := csiStream.Recv()
+			if err == io.EOF {
+				done <- true
+				return
+			}
+			if err != nil {
+				errCh <- fmt.Errorf(msgInternalFailedCSIDriverResponseFmt, err)
+				return
+			}
+			clientResp := convertToGetMetadataAllocatedResponse(csiResp)
+			if err := stream.Send(clientResp); err != nil {
+				errCh <- fmt.Errorf(msgInternalFailedtoSendResponseFmt, err)
+				return
+			}
+		}
```

Comment:
Consider putting the for loop in a helper method. It is easier to follow in the handler and will definitely be easier to unit test.  e.g.
```
func (s *Server) streamGetMetadataAllocatedResponse(... args ) error {
   for {
      ...
   }
}
```